Local WebKit builds don't use locally-built WebKit process extensions

pvollan opened the pull request:
#### 2bc6c250ae104d10775fe6240906378d997313c0
<pre>
Local WebKit builds don&apos;t use locally-built WebKit process extensions
<a href="https://bugs.webkit.org/show_bug.cgi?id=273550">https://bugs.webkit.org/show_bug.cgi?id=273550</a>
<a href="https://rdar.apple.com/125726458">rdar://125726458</a>

Reviewed by Elliott Williams.

When locally built WebKit extensions are installed in the app bundle, these extensions
will be launched instead of the system extensions when launching said app. This patch
adds a build step that copies extensions to WebKitTestRunner app bundle.
When WebKitTestRunner contains browser extensions, it also needs the Web browser
entitlement. Adding this introduced a couple of test failures related to in-app
browsers. These failures were resolved by adding a bundle ID check for the custom
bundle ID that is set in these tests.

* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.h:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::treatAsNonBrowser):
(WebKit::isParentProcessAFullWebBrowser):
(WebKit::isFullWebBrowserOrRunningTest):
(WebKit::isInWebKitChildProcess):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::hasExtensionInAppBundle):
(WebKit::launchWithExtensionKit):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/280387@main">https://commits.webkit.org/280387@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6b98e4cc7df26e19b396d8d3d19b71dfb21dc6eb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56388 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35714 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8860 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/59995 "Hash 6b98e4cc for PR 30101 does not build (failure)") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6824 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58514 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43336 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7018 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/59995 "Hash 6b98e4cc for PR 30101 does not build (failure)") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4768 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58417 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33589 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48654 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/59995 "Hash 6b98e4cc for PR 30101 does not build (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30369 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5984 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/5828 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52355 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6256 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61679 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/296 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6378 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52939 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/296 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48720 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/52824 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/264 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8387 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31541 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32627 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33710 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32374 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
```diff
@@ -273,7 +286,7 @@ bool isFullWebBrowserOrRunningTest(const String& bundleIdentifier)
 #endif
 
-    return fullWebBrowser || isRunningTest(bundleIdentifier);
+    return (fullWebBrowser && !isRunningInAppBrowserTest(bundleIdentifier)) || isRunningTest(bundleIdentifier);
```
I am a bit confused about this. Who is setting the inappbrowsertest bundle identifier?

pvollan replied:
That is set in the in-app browser tests under http/tests/in-app-browser-privacy

Thanks for reviewing!

szewai reviewed (approved):
r=me if bots are happy

pvollan committed:
Local WebKit builds don't use locally-built WebKit process extensions
https://bugs.webkit.org/show_bug.cgi?id=273550
rdar://125726458

Reviewed by Elliott Williams.

When locally built WebKit extensions are installed in the app bundle, these extensions
will be launched instead of the system extensions when launching said app. This patch
adds a build step that copies extensions to WebKitTestRunner app bundle.
When WebKitTestRunner contains browser extensions, it also needs the Web browser
entitlement. Adding this introduced a couple of test failures related to in-app
browsers. These failures were resolved by adding a bundle ID check for the custom
bundle ID that is set in these tests.

* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.h:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::treatAsNonBrowser):
(WebKit::isParentProcessAFullWebBrowser):
(WebKit::isFullWebBrowserOrRunningTest):
(WebKit::isInWebKitChildProcess):
* Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm:
(WebKit::hasExtensionInAppBundle):
(WebKit::launchWithExtensionKit):
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp-iOS.entitlements:
* Tools/WebKitTestRunner/Configurations/WebKitTestRunnerApp.xcconfig:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/280387@main
